wts/api.py

```diff
@@ -78,19 +78,32 @@ def load_settings(app):
             state_prefix = ""
 
         for idp, idp_conf in conf.get("login_options", {}).items():
-            authorization_url = fence_base_url + "oauth2/authorize"
-            authorization_url = add_params_to_uri(
-                authorization_url, idp_conf.get("params", {})
-            )
+            scope = "openid data user"
+            idp_params = idp_conf.get("params", {})
+            idp_client = idp_params.get("idp", "")
+            if idp_client == "keycloak":
+                auth_endpoint = idp_params.get("auth_url")
+                token_endpoint = idp_params.get("token_url")
+                scope = idp_params.get("scope", scope)
+                authorization_url = url + auth_endpoint + "?idp=" + idp_client
+                access_token_url = url + token_endpoint
```

Comment:
- `url + auth_endpoint` and `url + token_endpoint`: we should use urllib urljoin to build the url (or manually make sure there are slashes where you expect them)
- `+ "?idp="`: this won't work if the url already contains query parameters. i think urllib has tools to add a parameter. Or maybe we can use the `add_params_to_uri` function that's already used elsewhere in this file

Reply:
i see the existing code works is written this way too... it would be cool to update it and make it more resilient too